### PR TITLE
Adopt idiomatic Kotlin in RecordedRequest

### PR DESCRIPTION
**Pull request by swankjesse**
Not much to change.

### CODE REVIEW

File: mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt [hunk 1]
```diff
@@ -105,12 +115,16 @@ class RecordedRequest(
...
 
+  @Deprecated(
+      message = "Use body.readUtf8()",
+      replaceWith = ReplaceWith("body.readUtf8()"),
+      level = DeprecationLevel.WARNING)
+  fun getUtf8Body(): String = body.readUtf8()
+
   /** Returns the first header named [name], or null if no such header exists. */
   fun getHeader(name: String): String? {
     return headers.values(name).firstOrNull()
```
Comment (yschimke) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt]: nit: could return expression like below also

File: mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt [hunk 2]
```diff
@@ -105,12 +115,16 @@ class RecordedRequest(
     }
   }
 
+  @Deprecated(
+      message = "Use body.readUtf8()",
+      replaceWith = ReplaceWith("body.readUtf8()"),
```
Comment (yschimke) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt]: curious: Does Intellij also convert the kotlin suggestions to Java?

Comment (swankjesse) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt]: No, Java users have to fix manually.

File: mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt (hunk 1 above)
Comment (swankjesse) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt]: Fixed!

**Pull request merged**